src/tiled/abstractobjecttool.cpp

```diff
@@ -358,13 +358,20 @@ void AbstractObjectTool::showContextMenu(MapObjectItem *clickedObjectItem,
 
         auto changeTileAction = menu.addAction(tr("Replace Tile"), this, SLOT(changeTile()));
         changeTileAction->setEnabled(tile());
-
-        auto replaceTemplateAction = menu.addAction(tr("Replace With Template"), this, SLOT(replaceObjectsWithTemplate()));
-        replaceTemplateAction->setEnabled(objectTemplate());
     }
 
+    // Create action for replacing an object with a template
+    auto selectedTemplate = objectTemplate();
+    auto replaceTemplateAction = menu.addAction(tr("Replace With Template"), this, SLOT(replaceObjectsWithTemplate()));
+
+    if (selectedTemplate)
+        replaceTemplateAction->setText(tr("Replace With \"") + selectedTemplate->name() + tr("\" Template"));
```

Comment:
This is not nice for the translator. You should instead use a parameter, like so:
```c++
tr("Replace With \"%1\" Template").arg(selectedTemplate->name())
```

Reply:
Alright.